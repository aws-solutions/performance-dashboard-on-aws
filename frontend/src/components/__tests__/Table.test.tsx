/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import dayjs from "dayjs";
import { render, screen, fireEvent } from "@testing-library/react";
import Table from "../Table";

const columns = [
    {
        Header: "ID",
        accessor: "id",
    },
    {
        Header: "Name",
        accessor: "name",
    },
    {
        Header: "Last Updated",
        accessor: "updatedAt",
        Cell: (props: any) => {
            return dayjs(props.value).format("YYYY-MM-DD");
        },
    },
];

const rows = [
    {
        id: "1",
        name: "Banana",
        updatedAt: "2021-11-11",
    },
    {
        id: "2",
        name: "Chocolate",
        updatedAt: "2020-11-11",
    },
    {
        id: "3",
        name: "Vanilla",
        updatedAt: "2019-11-11",
    },
];

test("renders a basic table", async () => {
    const { container } = render(<Table selection="none" columns={columns} rows={rows} />);
    expect(container).toMatchSnapshot();
});

test("renders a table with selection checkboxes", async () => {
    render(
        <Table
            selection="multiple"
            columns={columns}
            rows={rows}
            screenReaderField="name"
            rowTitleComponents={["name", "updatedAt"]}
        />,
    );
    expect(
        screen.getByRole("checkbox", {
            name: "Banana - 2021-11-11",
        }),
    ).toBeInTheDocument();

    expect(
        screen.getByRole("checkbox", {
            name: "Chocolate - 2020-11-11",
        }),
    ).toBeInTheDocument();

    expect(
        screen.getByRole("checkbox", {
            name: "Vanilla - 2019-11-11",
        }),
    ).toBeInTheDocument();
});

test("calls onSelection function when user selects a row", async () => {
    const onSelection = jest.fn();
    render(
        <Table
            selection="multiple"
            columns={columns}
            rows={rows}
            screenReaderField="name"
            onSelection={onSelection}
        />,
    );

    const checkbox = screen.getByRole("checkbox", { name: "Chocolate" });
    fireEvent.click(checkbox);

    expect(onSelection).toBeCalledWith([
        {
            id: "2",
            name: "Chocolate",
            updatedAt: "2020-11-11",
        },
    ]);
});

test("sorting buttons are clickable", async () => {
    const { container } = render(
        <Table
            selection="multiple"
            columns={columns}
            rows={rows}
            screenReaderField="name"
            initialSortByField="name"
        />,
    );

    // Sort by ascending order
    fireEvent.click(
        screen.getByRole("button", {
            name: "Toggle SortBy: Last Updated",
        }),
    );

    // Vanilla should be first item
    expect(container).toMatchSnapshot();

    // Toggle to descending
    fireEvent.click(
        screen.getByRole("button", {
            name: "Toggle SortBy: Last Updated",
        }),
    );

    // Vanilla should be last
    expect(container).toMatchSnapshot();
});

test("filters out data when filterQuery is provided", async () => {
    render(
        <Table
            selection="none"
            columns={columns}
            rows={rows}
            screenReaderField="name"
            filterQuery="Chocolate"
        />,
    );

    expect(screen.getByText("Chocolate")).toBeInTheDocument();
    expect(screen.queryByText("Vanilla")).not.toBeInTheDocument();
    expect(screen.queryByText("Banana")).not.toBeInTheDocument();
});

test("renders a basic table without pagination", async () => {
    const { container } = render(
        <Table selection="none" columns={columns} rows={rows} disablePagination={true} />,
    );
    expect(container).toMatchSnapshot();
});

test("renders a basic table with mobile navigation", async () => {
    const { container } = render(
        <Table
            selection="none"
            columns={columns}
            rows={rows}
            disablePagination={true}
            mobileNavigation
        />,
    );
    expect(container).toMatchSnapshot();
});
