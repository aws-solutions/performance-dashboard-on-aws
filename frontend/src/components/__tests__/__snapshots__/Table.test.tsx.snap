// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic table 1`] = `
<div>
  <div
    class="overflow-x-hidden overflow-y-hidden"
  >
    <table
      class="usa-table usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: ID"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: ID"
              type="button"
            >
              <span>
                ID
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Name"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Name"
              type="button"
            >
              <span>
                Name
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Last Updated"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Last Updated"
              type="button"
            >
              <span>
                Last Updated
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            1
          </th>
          <td
            role="cell"
          >
            Banana
          </td>
          <td
            role="cell"
          >
            2021-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            role="cell"
          >
            Chocolate
          </td>
          <td
            role="cell"
          >
            2020-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            3
          </th>
          <td
            role="cell"
          >
            Vanilla
          </td>
          <td
            role="cell"
          >
            2019-11-11
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`renders a basic table with mobile navigation 1`] = `
<div>
  <div
    class="overflow-x-hidden overflow-y-hidden"
  >
    <table
      class="usa-table usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: ID"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: ID"
              type="button"
            >
              <span>
                ID
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Name"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Name"
              type="button"
            >
              <span>
                Name
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Last Updated"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Last Updated"
              type="button"
            >
              <span>
                Last Updated
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            1
          </th>
          <td
            role="cell"
          >
            Banana
          </td>
          <td
            role="cell"
          >
            2021-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            role="cell"
          >
            Chocolate
          </td>
          <td
            role="cell"
          >
            2020-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            3
          </th>
          <td
            role="cell"
          >
            Vanilla
          </td>
          <td
            role="cell"
          >
            2019-11-11
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`renders a basic table without pagination 1`] = `
<div>
  <div
    class="overflow-x-hidden overflow-y-hidden"
  >
    <table
      class="usa-table usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: ID"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: ID"
              type="button"
            >
              <span>
                ID
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Name"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Name"
              type="button"
            >
              <span>
                Name
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="min-width: 0;"
          >
            <button
              aria-label="Toggle SortBy: Last Updated"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Last Updated"
              type="button"
            >
              <span>
                Last Updated
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            1
          </th>
          <td
            role="cell"
          >
            Banana
          </td>
          <td
            role="cell"
          >
            2021-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            role="cell"
          >
            Chocolate
          </td>
          <td
            role="cell"
          >
            2020-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <th
            role="cell"
            scope="row"
          >
            3
          </th>
          <td
            role="cell"
          >
            Vanilla
          </td>
          <td
            role="cell"
          >
            2019-11-11
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`sorting buttons are clickable 1`] = `
<div>
  <div
    class="overflow-x-hidden overflow-y-hidden"
  >
    <table
      class="usa-table usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <span>
              <input
                style="cursor: pointer;"
                title="Toggle All Rows Selected"
                type="checkbox"
              />
            </span>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: ID"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: ID"
              type="button"
            >
              <span>
                ID
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: Name"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Name"
              type="button"
            >
              <span>
                Name
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="ascending"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: Last Updated"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Last Updated"
              type="button"
            >
              <span>
                Last Updated
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-circle-up fa-w-16 margin-left-1"
                data-icon="chevron-circle-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm231-113.9L103.5 277.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L256 226.9l101.6 101.6c9.4 9.4 24.6 9.4 33.9 0l17-17c9.4-9.4 9.4-24.6 0-33.9L273 142.1c-9.4-9.4-24.6-9.4-34 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Vanilla"
              style="cursor: pointer;"
              title="Vanilla"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            3
          </td>
          <td
            role="cell"
          >
            Vanilla
          </td>
          <td
            role="cell"
          >
            2019-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Chocolate"
              style="cursor: pointer;"
              title="Chocolate"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            2
          </td>
          <td
            role="cell"
          >
            Chocolate
          </td>
          <td
            role="cell"
          >
            2020-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Banana"
              style="cursor: pointer;"
              title="Banana"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            1
          </td>
          <td
            role="cell"
          >
            Banana
          </td>
          <td
            role="cell"
          >
            2021-11-11
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`sorting buttons are clickable 2`] = `
<div>
  <div
    class="overflow-x-hidden overflow-y-hidden"
  >
    <table
      class="usa-table usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <span>
              <input
                style="cursor: pointer;"
                title="Toggle All Rows Selected"
                type="checkbox"
              />
            </span>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: ID"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: ID"
              type="button"
            >
              <span>
                ID
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="none"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: Name"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Name"
              type="button"
            >
              <span>
                Name
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-down fa-w-14 margin-left-1"
                data-icon="chevron-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
          <th
            aria-sort="descending"
            colspan="1"
            role="columnheader"
            scope="col"
            style="padding: 0.5rem 1rem;"
          >
            <button
              aria-label="Toggle SortBy: Last Updated"
              class="usa-button usa-button--unstyled text-no-underline"
              style="cursor: pointer;"
              title="Toggle SortBy: Last Updated"
              type="button"
            >
              <span>
                Last Updated
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-circle-down fa-w-16 margin-left-1"
                data-icon="chevron-circle-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zM273 369.9l135.5-135.5c9.4-9.4 9.4-24.6 0-33.9l-17-17c-9.4-9.4-24.6-9.4-33.9 0L256 285.1 154.4 183.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L239 369.9c9.4 9.4 24.6 9.4 34 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Banana"
              style="cursor: pointer;"
              title="Banana"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            1
          </td>
          <td
            role="cell"
          >
            Banana
          </td>
          <td
            role="cell"
          >
            2021-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Chocolate"
              style="cursor: pointer;"
              title="Chocolate"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            2
          </td>
          <td
            role="cell"
          >
            Chocolate
          </td>
          <td
            role="cell"
          >
            2020-11-11
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            role="cell"
          >
            <input
              aria-labelledby="Vanilla"
              style="cursor: pointer;"
              title="Vanilla"
              type="checkbox"
            />
          </td>
          <td
            role="cell"
          >
            3
          </td>
          <td
            role="cell"
          >
            Vanilla
          </td>
          <td
            role="cell"
          >
            2019-11-11
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
